analyzer/src/main/kotlin/curation/ClearlyDefinedPackageCurationProvider.kt

```diff
@@ -110,12 +105,8 @@ class ClearlyDefinedPackageCurationProvider(
             }
         }.toMap()
 
-        val contributedCurations = runCatching {
-            buildMap {
-                coordinatesToIds.keys.chunked(BULK_REQUEST_SIZE).forEach { coordinates ->
-                    putAll(runBlocking(Dispatchers.IO) { service.getCurations(coordinates) })
-                }
-            }
+        val curations = runCatching {
+            runBlocking(Dispatchers.IO) { service.getCurationsChunked(coordinatesToIds.keys) }
```

Comment:
Can you help me understanding this please? 
I understand that the single call to `getCurationsChunked()` is scheduled / executed on another thread, and this thread blocks and waits for it to return. I wonder what's the reason to do that context switch....

Reply:
The original `forEach` loop from this code location has been moved to `getCurationsChunked`, but *without* the `runBlocking``around the inner `getCurations` suspend function call. So as `getCurationsChunked` now calls a suspend function, it needs to be a suspend function itself, unless it uses `runBlocking` itself. The reason why I chose not to make `getCurationsChunked` use `runBlocking` itself is that it's usually a good idea to block as late as possible, i.e. the caller should decide whether it need the result now, or if it could wait a bit as it does more other things in parallel.

Reply:
Maybe @oheger-bosch, any opinion here?

Reply:
It makes sense for `getCurationsChunked` to be a suspend function because the rest of the `ClearlyDefinedService` API is also suspend functions. So we have to use `runBlocking`here because `getCurationsFor` is not a suspend function.
Regarding `Dispatchers.IO` it is just best practice to use it for IO calls. It's of limited use here, because the current thread is blocked anyway. For example, it would make more sense to use it if this code was a coroutine running in a Dispatcher with limited threads, like the UI thread. However, according to the [docs ](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-dispatchers/-i-o.html), the use of `Dispatchers.IO` here might not even lead to a context switch:

> This dispatcher and its views share threads with the [Default](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-dispatchers/-default.html) dispatcher, so using withContext(Dispatchers.IO) { ... } when already running on the [Default](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-dispatchers/-default.html) dispatcher typically does not lead to an actual switching to another thread. In such scenarios, the underlying implementation attempts to keep the execution on the same thread on a best-effort basis.

Reply:
So, anyone dares to approve?